test/e2e/framework/addon/globals.go

```diff
@@ -61,21 +63,44 @@ func InitGlobals(cfg *config.Config) {
 	}
 	globalsInited = true
 	*Base = base.Base{}
+	*Vault = vault.Vault{
+		Base:      Base,
+		Namespace: "e2e-vault",
+		Name:      "vault",
+	}
 	allAddons = []Addon{
 		Base,
+		Vault,
 	}
 }
 
-// ProvisionGlobals provisions all of the global addons, including calling Setup.
+// SetupGlobals setups all of the global addons.
 // This should be called by the test suite entrypoint in a SynchronizedBeforeSuite
-// block to ensure it is run once per suite.
-func ProvisionGlobals(cfg *config.Config) error {
-	// TODO: if we want to provision dependencies in parallel we will need
-	// to improve the logic here.
-	for _, g := range allAddons {
-		if err := provisionGlobal(g, cfg); err != nil {
+// block to ensure it is run once per suite (only on ginkgo process #1).
+func SetupGlobalsPrimary(cfg *config.Config) ([]interface{}, error) {
+	toBeTransferred := make([]interface{}, len(allAddons))
+	for idx, g := range allAddons {
+		data, err := g.Setup(cfg)
+		if err != nil {
+			return nil, err
+		}
+		if !g.SupportsGlobal() {
+			return nil, fmt.Errorf("requested global plugin does not support shared mode with current configuration")
+		}
+		toBeTransferred[idx] = data
```

Comment:
Can we make []interface{} into map[AddonName]interface{} and use addon name as a key? (Add a `Name` method  to addons)

Reply:
I changed it to a `map[string]interface{}` value.

Reply:
I think it is easier to keep this as a statically allocated array, otherwise we have to add logic for keys which are not that useful.